rust/src/tests/errors/eexist.rs

```diff
@@ -0,0 +1,107 @@
+use nix::{
+    errno::Errno,
+    fcntl::{open, OFlag},
+    sys::stat::{mknod, Mode, SFlag},
+    unistd::{mkdir, mkfifo},
+};
+
+#[cfg(any(
+    target_os = "openbsd",
+    target_os = "netbsd",
+    target_os = "freebsd",
+    target_os = "dragonfly",
+    target_os = "macos",
+    target_os = "ios"
+))]
+use nix::{sys::stat::FileFlag, unistd::chflags};
+
+use std::{fmt::Debug, path::Path};
+
+use crate::{
+    runner::context::{FileType, TestContext},
+    utils::{link, rename, rmdir, symlink},
+};
+
+/// Asserts that it returns EEXIST if the named file exists
+fn assert_eexist<F, T: Debug>(ctx: &mut TestContext, ft: &FileType, f: F)
+where
+    F: Fn(&Path) -> nix::Result<T>,
+{
+    let path = ctx.create(ft.clone()).unwrap();
+    assert_eq!(f(&path).unwrap_err(), Errno::EEXIST);
+}
+
+crate::test_case! {
+    /// Return EEXIST if the named file exists
+    named_file => [Regular, Dir, Fifo, Block, Char, Socket, Symlink(None)]
+}
+fn named_file(ctx: &mut TestContext, ft: FileType) {
+    //TODO: Add rmdir/{06,12}.t
+
+    let default_mode = Mode::from_bits_truncate(0o644);
+
+    // mkdir/10.t
+    assert_eexist(ctx, &ft, |p| mkdir(p, Mode::empty()));
+
+    // mknod/08.t
+    assert_eexist(ctx, &ft, |p| mknod(p, SFlag::S_IFIFO, Mode::empty(), 0));
```

Comment:
These different assertions really shouldn't be in the same test case.  They need to be split into separate test cases.

Reply:
Which ones? Those which don't create files?

Reply:
I mean that mkdir should get one test case, mknod another, mkfifo a third, etc.